scripts/upgrade

```diff
@@ -101,25 +103,63 @@ then
   if 	[ -d "$backup_squelettes" ]
   then
     cp -ar $backup_squelettes $final_path/www/squelettes
-    ynh_secure_remove  $backup_squelettes
-  fi
+   fi
 
   if [ -e "$backup_bdd" ]
   then
 	  cp -a $backup_bdd $final_path/association.sqlite
-    ynh_secure_remove $backup_bdd
   fi
 
   chown -R $app:$app $final_path
+  chmod 755 $final_path
+
+  # Create the visitors permission if needed
+   visitors_enabled=$(ynh_permission_has_user "main" "visitors" && echo 0 || echo 1)
+  if [[ $visitors_enabled -eq 0 ]]; then
+    ynh_permission_update --permission "main" --add "visitors"
+  fi
+
+  ynh_systemd_action --service_name=nginx --action=reload
+  # Finalize the upgrade
+  ynh_local_curl "/index.php"
+  sleep 5
+  ynh_local_curl "/index.php"
+  upgrade_type="UPGRADE_APP"
+
+  if [[ $visitors_enabled -eq 0 ]]; then
+    ynh_permission_update --permission "main" --remove "visitors"
+  fi
+fi
+#=================================================
+# UPGRADE TO 1.1.0 if necessary
+#=================================================
 
-  key=$(ynh_string_random --length=50)
-  cp -a $final_path/config.dist.php $final_path/config.local.php
-  ynh_replace_string --match_string="const SECRET_KEY = '3xUhIgGwuovRKOjVsVPQ5yUMfXUSIOX2GKzcebsz5OINrYC50r';" --replace_string="const SECRET_KEY = '$key';" --target_file="$final_path/config.local.php"
-  if [ "$path_url" == "/" ]; then
-  ynh_replace_string --match_string="//const WWW_URI = '/asso/';" --replace_string="const WWW_URI = '$path_url';" --target_file="$final_path/config.local.php"
-  else
-  ynh_replace_string --match_string="//const WWW_URI = '/asso/';" --replace_string="const WWW_URI = '$path_url/';" --target_file="$final_path/config.local.php"
+if [ "$upgrade_type" == "1.1.0" ]
```

Comment:
Not sure to understand why you try to define `upgrade_type` earlier rather than just writing here :  

```
if ynh_compare_current_package_version --comparison lt --version 1.1.0
```

(note that I used a strict lt / "lower than" here which allows to just express "if version < 1.1.0" which is easier to write and readd)

Same thing should be applied for the 0.9.8 block I believe

Reply:
I thought it was needed with `ynh_compare_current_package_version --comparison` to explain a version 
which have been used  by Yunohost before like x.x.x~ynh1 or x.x.x~ynh2. I said this because we didn't have done a version 1.1.0~ynh, we just jump from 0.9.8.1 to 1.0.2 and now to 1.1.11 or 1.1.12. So if I use this expression `if ynh_compare_current_package_version --comparison lt --version 1.1.0` do you think the expression is enough to understand and look for the garradin's version ??

Reply:
Yes, as said this will just mean "if the version we are upgrading from is strictly lower than 1.1.0" (which is what we want).

Reply:
Ok, I test your suggestion...

Reply:
It fails doing this, look at https://paste.yunohost.org/raw/uzaxunaber

Reply:
`2021-11-01 16:24:36,404: DEBUG - + [[ ! 0.9.8.1 =~ ~ynh ]]
2021-11-01 16:24:36,404: DEBUG - + ynh_die '--message=Invalid argument for version.'
2021-11-01 16:24:36,423: DEBUG - + ret_code=1
2021-11-01 16:24:36,423: DEBUG - + echo 'Invalid argument for version.'
2021-11-01 16:24:36,423: DEBUG - + exit 1`

Reply:
So I think is better keep the code as before which works. Do you agree ?

Reply:
doing this, I also couldn't restore the backup garradin-pre-upgrade1 created before it fails...

Reply:
Sorry, I forgot telling you before I tested yours indications on the testing branch, but come back to the previous lines as it fails...

Reply:
Hmpf it seems to be expecting a version in the "yunohost format" so you gotta add `~ynh1` for example:

```
ynh_compare_current_package_version --comparison lt --version 0.9.8.1~ynh1
```

Reply:
Ok, and if I use this if `ynh_compare_current_package_version --comparison lt --version 1.1.0~yhh1` it will works however this version exist ?

Reply:
Ok, I have try this and it reproduce the error, after the upgrade successfull I loose my datas and arrive on the install.php page !

Reply:
The code before was more efficient, I don't know why but it resolve the issue...

Reply:
Ok, I see a mistake, `if ynh_compare_current_package_version --comparison lt --version 0.1.1.0~ynh1` I try again...

Reply:
It works ! thanks